WordPress/src/main/java/org/wordpress/android/ui/stories/prefs/StoriesPrefs.kt

```diff
@@ -0,0 +1,128 @@
+package org.wordpress.android.ui.stories.prefs
+
+import android.content.Context
+import android.net.Uri
+import androidx.preference.PreferenceManager
+import com.wordpress.stories.compose.story.StoryFrameItem
+import com.wordpress.stories.compose.story.StoryFrameItem.BackgroundSource.FileBackgroundSource
+import com.wordpress.stories.compose.story.StoryFrameItem.BackgroundSource.UriBackgroundSource
+import com.wordpress.stories.compose.story.StorySerializerUtils
+
+object StoriesPrefs {
+    private const val KEY_PREFIX_STORIES_SLIDE_ID = "story_slide_id-"
+    private const val KEY_PREFIX_LOCAL_MEDIA_ID = "l-"
+    private const val KEY_PREFIX_REMOTE_MEDIA_ID = "r-"
+
+    private fun buildSlideKey(siteId: Long, mediaId: RemoteMediaId): String {
+        return KEY_PREFIX_STORIES_SLIDE_ID + siteId.toString() + "-" +
+                KEY_PREFIX_REMOTE_MEDIA_ID + mediaId.mediaId.toString()
+    }
+
+    private fun buildSlideKey(siteId: Long, mediaId: LocalMediaId): String {
+        return KEY_PREFIX_STORIES_SLIDE_ID + siteId.toString() + "-" +
+                KEY_PREFIX_LOCAL_MEDIA_ID + mediaId.id.toString()
+    }
+
+    @JvmStatic
+    fun checkSlideIdExists(context: Context, siteId: Long, mediaId: RemoteMediaId): Boolean {
+        val slideIdKey = buildSlideKey(siteId, mediaId)
+        return PreferenceManager.getDefaultSharedPreferences(context).contains(slideIdKey)
+    }
+
+    @JvmStatic
+    fun checkSlideOriginalBackgroundMediaExists(context: Context, siteId: Long, mediaId: RemoteMediaId): Boolean {
+        val storyFrameItem: StoryFrameItem? = getSlideWithRemoteId(context, siteId, mediaId)
+        storyFrameItem?.let { frame ->
+            // now check the background media exists or is accessible on this device
+            frame.source.let { source ->
+                if (source is FileBackgroundSource) {
+                    source.file?.let {
+                        return it.exists()
+                    } ?: return false
+                } else if (source is UriBackgroundSource) {
+                    source.contentUri?.let {
+                        return isUriAccessible(it, context)
+                    } ?: return false
+                }
+            }
+        }
+        return false
+    }
+
+    private fun isUriAccessible(uri: Uri, context: Context): Boolean {
+        if (uri.toString().startsWith("http")) {
+            // TODO: assume it'll be accessible - we'll figure out later
+            // potentially force external download using MediaUtils.downloadExternalMedia() here to ensure
+            return true
+        }
+        try {
+            val inputStream = context.contentResolver.openInputStream(uri)
+            if (inputStream != null) {
+                inputStream.close()
+                return true
+            }
+        } catch (e: java.lang.Exception) {
+            e.printStackTrace()
+        }
+        return false
+    }
```

Comment:
Nitpick : This could probably be simplified like so 😉 
```suggestion
        context.contentResolver.openInputStream(uri).use { stream -> 
            return stream != null
        }
```

Reply:
good idea; also noted we need to call `inputStream.close()` before returning so, ended up doing this in 34d4effeac:

```
            context.contentResolver.openInputStream(uri)?.let {
                it.close()
                return true
            }
```

Reply:
nice. Good going 👍